### DISCUSSION TITLE
Adding new painting or image

### DISCUSSION THREAD

**Discussion by JACKWANGU158** [allenai/ai2thor#1106] [Q&A]
I want to create a new image as the target to navigate. How can I create or replace an image through Unity.

**Comment by elimvb** [comment]
What kind of image do you want to place in Unity? Is it square, or rectangular, because with non-square images the process has an extra step in the setup.

**Reply by JACKWANGU158** [reply]
Thank you for your reply. I want to add a photo to the wall of the room, which shows the cracks and spalling of the concrete. Replace the oil painting in the room with an RGB image.

**Comment by elimvb** [comment]
1. Look up the resolution of the image you want to add. You'll need that for later.
![Test_Pattern](https://user-images.githubusercontent.com/40370237/202298846-5495fe4d-9cb5-4b8e-87ae-51ad41a3bcd5.jpg)
2. In Unity, once you're in the scene file you want to add the image to (in this example, FloorPlan1) in Unity Editor, create a Quad GameObject from the **GameObject** dropdown menu.
![1 - Create_Quad](https://user-images.githubusercontent.com/40370237/202300803-2c7123c6-4cb1-4600-936a-6a46793bad92.jpg)
![2 - Quad](https://user-images.githubusercontent.com/40370237/202300782-ebbc0ba8-2bee-4db2-bb10-9bc039d7450e.jpg)
3. In order to display your unique bitmap to this quad, you need a new "material", which is a set of visual rules that defines how an object renders (color, shininess, self-illumination, etc.). I understand you just want a simple projection of the image, so, in your "Project Explorer" window in the Unity Editor, right-click and select "Create > Material" to create a default material, then move it to wherever you desire in the project (in this example, I simply have it in the "Assets" folder), and rename it to whatever you want (in this example, "Test_Pattern_Material".
![3 - Create_Material](https://user-images.githubusercontent.com/40370237/202303039-2f1a373a-f370-40f9-8d90-ce85f44c53fc.jpg)
![4 - Material](https://user-images.githubusercontent.com/40370237/202303051-ad8da84f-5776-4b23-9382-959c4b50948d.jpg)
4. Move the bitmap file to the same location as your new material.
![5 - Image](https://user-images.githubusercontent.com/40370237/202304684-cd73a9a1-7337-4c6f-bbd9-0fb6c538707e.jpg)
5. In Unity Editor, drag your bitmap onto the "Albedo" (i.e. "Color") channel of the new Unity Material, and the bitmap will be applied to the color channel of that material.
![6---texture_to_material_small](https://user-images.githubusercontent.com/40370237/202306557-1fc91723-4847-4cda-a115-6c11ee741f84.gif)
6. Now drag your Unity Material onto the quad, and the material (with your bitmap) will now be on the surface.
![7---material_to_object_small](https://user-images.githubusercontent.com/40370237/202306550-a94f6bd7-87b4-465f-b779-4c39902cee41.gif)
7. Finally, if your image is rectangular, you'll want to match the dimensions of the quad to the aspect ratio of the image. Since the example image I used has an aspect ratio of 1280:960, in the Inspector window of the Unity Editor, set the "Scale" of the X and Y axes to "1.26" and "0.96", and now you will have an immaculate render of your bitmap in Unity!
![8 - Scale](https://user-images.githubusercontent.com/40370237/202307534-58f0263a-52f5-4ff0-a0be-5c33f4c54ff2.jpg)
8. If you want to scale the bitmap up or down, use the white cube of the Scale gizmo to resize it manually.
![9---scale_manually_small](https://user-images.githubusercontent.com/40370237/202308421-967acf58-fae7-4911-a1c9-b074a0f5200f.gif)

**Reply by JACKWANGU158** [reply]
Thank you for your detailed reply. I also want to know how to establish the interaction between ai2thor and Unity, such as how to download the scene files in ai2thor. I tried to do it but failed.

**Reply by elimvb** [reply]
Are you talking about being able to access Unity Editor, to edit the scenes and stuff?

**Reply by JACKWANGU158** [reply]
Happy New Year! Thanks for your reply. I noticed that in your code, the h5 file is used for scene import, but the scene I modified in unity is saved as a unity file. So how to convert the unity file into an h5 file?
